SYCL/Basic/half_type.cpp

```diff
@@ -192,7 +192,48 @@ inline bool bitwise_comparison_fp32(const half val, const uint32_t exp) {
   return reinterpret_cast<const uint32_t &>(fp32) == exp;
 }
 
+constexpr void constexpr_verify_add() {
+  constexpr half a{5.0}, b{2.0}, ref{7.0};
+  constexpr half result = a + b;
+  constexpr half diff = result - ref;
+  static_assert(std::fabs(static_cast<float>(diff)) <
+                    std::numeric_limits<cl::sycl::half>::epsilon(),
+                "Constexpr add is wrong");
```

Comment:
This could be refactored to a function:

```suggestion
  static_assert(is_close(ref, result), "Constexpr add is wrong");
```

```cpp
constexpr bool is_close(sycl::half a, sycl::half b) {
  constexpr half diff = a - b;
  return std::fabs(static_cast<float>(diff) < std::numeric_limits<sycl::half>::epsilon());
}
```